### PR TITLE
Add code tabs to fun-write-method-returns-function

### PR DESCRIPTION
**Pull request by doehyunbaek**
With c875c656a62e1a07cd09bdb5eb6d7650c3ca90ab I updated Gemfile.lock.

Is it needed or better not tracked in git? I assumed the former from the previous commits to the file and if i misunderstood it I'll revert and force push the change.

With 60a87c4d20434a17fea9cf1ba0bb9c24ca2a36ab, I updated code tab, per #2481.

I tested each code with 2.13.10 repl and 3.2.1 repl and tested the website with jekyll

### CODE REVIEW

File: _overviews/scala3-book/fun-write-method-returns-function.md [hunk 1]
```diff
@@ -116,27 +156,53 @@ A first thing you know is that you want to create a method that (a) takes a “d
...
 val frenchGreeting = (name: String) => println(s"Bonjour, $name")
 ```
+{% endtab %}
+{% endtabs %}
 
 Now all you need to do is (a) return `englishGreeting` if the `desiredLanguage` is English, and (b) return `frenchGreeting` if the `desiredLanguage` is French.
 One way to do that is with a `match` expression:
 
+{% tabs fun-write-method-returns-function-14 class=tabs-scala-version %}
+{% tab 'Scala 2' for=fun-write-method-returns-function-14 %}
```
Comment (julienrf) [inline _overviews/scala3-book/fun-write-method-returns-function.md:193]: We don’t need the `for` attribute anymore.

**Review by julienrf — APPROVED** [review]
Thank you very much @doehyunbaek for your help!

I just noticed one minor possible improvement.

File: _overviews/scala3-book/fun-write-method-returns-function.md (hunk 1 above)
Comment (doehyunbaek) [reply _overviews/scala3-book/fun-write-method-returns-function.md]: Done with [9651d75](https://github.com/scala/docs.scala-lang/pull/2664/commits/9651d757bd5163e9b1bbbc311117aaa908d8209d).

**Comment by doehyunbaek** [comment]
I added the for attribute because of [this mention](https://github.com/scala/docs.scala-lang/blob/008fc4beee8b0d83ddaba4c2e1eabbc56a953003/_overviews/contribute/add-guides.md#L153-L162) and should it also be updated?

**Comment by doehyunbaek** [comment]
I rebased onto origin/main and force-pushed branch.

File: Gemfile.lock [hunk 2]
```diff
@@ -91,4 +94,4 @@ DEPENDENCIES
   webrick (~> 1.7)
 
 BUNDLED WITH
-   2.3.10
+   2.4.2
```
Comment (julienrf) [inline Gemfile.lock:97]: I am not sure about this version change for the bundler. Here we use 2.3.10: https://github.com/scala/docs.scala-lang/blob/00302d7083b5320c54e65203cf7e16ee94fd3949/Dockerfile#L4

**Comment by julienrf** [comment]
Thank you for updating the PR. I don’t think the documentation should be updated, the advice explaining the possibility of adding the `for` property when that helps readability still applies.

File: Gemfile.lock (hunk 2 above)
Comment (doehyunbaek) [reply Gemfile.lock:97]: Can you give me a brief advice on when to commit Gemfile.lock file change? It seems always is not the option. So is it never? or for some portion of the lock file(e.g. except bundler version?

Comment (doehyunbaek) [reply Gemfile.lock]: Updated with [fc15ab3](https://github.com/scala/docs.scala-lang/pull/2664/commits/fc15ab3079958eb8354833bf980d39dab2f7be36), would still appreciate your advice.

Comment (julienrf) [reply Gemfile.lock]: The changes to the lockfile should always be committed, but the file should be generated with the right version of the bundler. Could you please make sure to have bundler 2.3.10 on your system, or to use the Docker approach to automatically use the correct version of the bundler?

Comment (doehyunbaek) [reply Gemfile.lock]: I actually have bundler 2.3.10 installed at my local mac. Strange it changed the lockfile. 

I will try to commit lockfile with correct bundler version as specified in Dockerfile and will use the Docker approach if I'm not able to do that.

**Pull request merged**